lib/datadog/tracing/tracer.rb

```diff
@@ -341,12 +347,13 @@ def bind_trace_events!(trace_op)
         events = trace_op.send(:events)
 
         events.span_before_start.subscribe do |event_span_op, event_trace_op|
-          event_trace_op.service ||= @default_service
```

Comment:
A small improvement here: we used to always set `@default_service` as the fallback service for a the newly created `TraceOperation`. By explicity setting the value of `event_trace_op.service` here, the `TraceOperation` can no longer use the root span service name as its default value, which is the intended behavior: https://github.com/DataDog/dd-trace-rb/blob/b3c5a3d57b7441da2d7c7ceb3e24acb4c5846bf5/lib/datadog/tracing/trace_operation.rb#L171-L173

This PR no longer explicitly sets the service field in the active `TraceOperation`, letting `event_trace_op.service` handle the fallback normally, as it was originally intended.